scripts/register-completions.zsh

```diff
@@ -1,10 +1,3 @@
-# zsh parameter completion for the dotnet CLI
-
-_dotnet_zsh_complete() 
-{
-  local completions=("$(dotnet complete "$words")")
-
-  reply=( "${(ps:\n:)completions}" )
-}
-
-compctl -K _dotnet_zsh_complete dotnet
+#compdef dotnet
+local completions=("$(dotnet complete "$words")")
+compadd "${(ps:\n:)completions}"
```

Comment:
Hey there, coming back to this to add that it should be
```zsh
compadd -- "${(ps:\n:)completions}"
```
to avoid dash options to be passed to `compadd` instead of being shown as completion entries.

Reply:
Also we should only call `compadd` if there are `$completions`, i.e. if `[[ -n "$completions" ]]` is true. Otherwise a blank space is called each time a completion is triggered and there are no entries returned.